src/Language/PureScript/TypeChecker/TypeSearch.hs

```diff
@@ -76,6 +78,32 @@ checkSubsume unsolved env st userT envT = checkInEnvironment env st $ do
   -- Finally, check any constraints which were found during elaboration
   Entailment.replaceTypeClassDictionaries (isJust unsolved) expP
 
+accessorSearch
+  :: Maybe [(P.Ident, Entailment.InstanceContext, P.Constraint)]
+  -> P.Environment
+  -> TC.CheckState
+  -> P.Type
+  -> ([(Label, Type)], [(Label, Type)])
+  -- ^ (all accessors we found, all accessors we found that match the result type)
+accessorSearch unsolved env st userT = maybe ([], []) fst $ checkInEnvironment env st $ do
+  let initializeSkolems =
+        Skolem.introduceSkolemScope
+        <=< P.replaceAllTypeSynonyms
+        <=< P.replaceTypeWildcards
+
+  userT' <- initializeSkolems userT
+
+  rowType <- freshType
+  resultType <- freshType
+  let recordFunction = TypeApp (TypeApp tyFunction (TypeApp tyRecord rowType)) resultType
+  _ <- subsumes recordFunction userT'
+  subst <- gets TC.checkSubstitution
+  let solvedRow = fst (rowToList (substituteType subst rowType))
+  tcS <- get
+  pure (solvedRow, filter (\x -> checkAccessor tcS (substituteType subst resultType) x) solvedRow)
+  where
+    checkAccessor tcs x (_, type') = isJust (checkSubsume unsolved env tcs x type')
```

Comment:
I thought about this some more... And running the subsumption checks for the labels within the same `StateT` computation would mean they influence one another, wouldn't it? In that case I prefer to keep them apart to make sure each is checked for fitness individually.